lib/Resque/Job.php

```diff
@@ -188,18 +191,49 @@ public function getInstance()
 	 */
 	public function perform()
 	{
+
+		$method = 'perform';
+		$beforePerformMethod = 'setUp';
+		$tearDownMethod = 'tearDown';
 		$instance = $this->getInstance();
+		if (is_array($instance))
+		{
+			if (isset($instance[4]))
```

Comment:
What is this sorcery?  These seem to be magic numbers.  I've no idea what these things mean :)